suite-native/fiat-rates/src/fiatRatesThunks.ts

```diff
@@ -0,0 +1,201 @@
+        currencies: [fiatCurrency],
+    });
+
+    const rate = success ? payload.tickers?.[0]?.rates?.[fiatCurrency] : null;
+    if (rate) return rate;
+
+    return fetchLastWeekFiatRates(ticker, fiatCurrency).then(
+        res => res?.tickers?.[0]?.rates?.[fiatCurrency],
+    );
+};
+
+const fetchFn: Record<RateType, typeof fetchFiatRate> = {
+    current: fetchFiatRate,
+    lastWeek: fetchLastWeekRate,
+};
+
+type UpdateCurrentFiatRatesThunkPayload = {
+    ticker: TickerId;
+    fiatCurrency: FiatCurrencyCode;
+    rateType?: RateType;
+};
+
+export const updateFiatRatesThunk = createThunk(
+    `${actionPrefix}/updateFiatRates`,
+    async ({ ticker, fiatCurrency, rateType = 'current' }: UpdateCurrentFiatRatesThunkPayload) => {
+        const rate = await fetchFn[rateType](ticker, fiatCurrency);
+
+        if (!rate) {
+            throw new Error('Failed to fetch fiat rates');
+        }
+
+        return rate;
+    },
+);
+
+export const updateMissingTxFiatRatesThunk = createThunk(
+    `${actionPrefix}/updateMissingTxRates`,
+    (
+        _,
+        {
+            dispatch,
+            getState,
+            extra: {
+                selectors: { selectLocalCurrency },
+            },
+        },
+    ) => {
+        const localCurrency = selectLocalCurrency(getState());
+        const transactionsWithMissingRates = selectTransactionsWithMissingRates(
+            getState(),
+            localCurrency,
+        );
+
+        transactionsWithMissingRates.forEach(({ account, txs }) => {
+            dispatch(updateTxsFiatRatesThunk({ account, txs, localCurrency }));
+        });
+    },
+);
+
+type FetchFiatRatesThunkPayload = {
+    rateType: RateType;
+};
+
+export const fetchFiatRatesThunk = createThunk(
+    `${actionPrefix}/fetchFiatRates`,
+    (
+        { rateType }: FetchFiatRatesThunkPayload,
+        {
+            dispatch,
+            getState,
+            extra: {
+                selectors: { selectLocalCurrency },
+            },
+        },
+    ) => {
+        const fiatCurrency = selectLocalCurrency(getState());
+        const tickers = selectTickersToBeUpdated(getState(), fiatCurrency, rateType);
+
+        return Promise.allSettled(
+            tickers.map(ticker =>
+                dispatch(
+                    updateFiatRatesThunk({
+                        ticker,
+                        fiatCurrency,
+                        rateType,
+                    }),
+                ),
+            ),
+        );
+    },
+);
+
+const ratesTimeouts: Record<RateType, ReturnType<typeof setTimeout> | null> = {
+    current: null,
+    lastWeek: null,
+};
+
+type PeriodicFetchFiatRatesThunkPayload = {
+    rateType: RateType;
+};
+
+export const periodicFetchFiatRatesThunk = createThunk(
+    `${actionPrefix}/periodicFetchFiatRates`,
+    async ({ rateType }: PeriodicFetchFiatRatesThunkPayload, { dispatch }) => {
+        if (ratesTimeouts[rateType]) {
+            clearTimeout(ratesTimeouts[rateType]!);
```

Comment:
```suggestion
            clearTimeout(ratesTimeouts[rateType]);
```

Shouldn't need the `!` because of the condition above?

Reply:
For some unknown reason it throws error without `!`...